hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -864,11 +867,55 @@ def __init__(
         self.size_one_allowed = self.min_side <= 1 <= self.max_side
 
     def do_draw(self, data):
+        # We don't usually have a gufunc signature; do the common case first & fast.
+        if self.signature is None:
+            return self._draw_loop_dimensions(data)
+
+        # When we *do*, draw the core dims, then draw loop dims, and finally combine.
+        core_in, core_res = self._draw_core_dimensions(data)
+
+        # If some core shape has ommited optional dimensions, it's an error to add
+        # loop dimensions to it.  We never omit core dims if min_dims >= 1.
```

Comment:
We should be sure to document this behavior!

Reply:
I'd rather not - we're simply avoiding generating certain shapes that we can't easily tell are valid by construction (ie. where the loop dims would be interepreted as bad core dims).  I've added a comment to this effect; but left it undocumented as we may fix it later.